### ISSUE TITLE
Generic binding : BindingFail status is reported 

### ISSUE THREAD

**Issue by sbose78** [redhat-developer/service-binding-operator#381]
Even though the binding secret gets created fine, the status is marked as `BindingFail`

```
apiVersion: apps.openshift.io/v1alpha1
kind: ServiceBindingRequest
metadata:
  creationTimestamp: '2020-03-12T21:05:44Z'
  generation: 1
  name: node-todo-git
  namespace: sbose
  resourceVersion: '221728'
  selfLink: >-
    /apis/apps.openshift.io/v1alpha1/namespaces/sbosse/servicebindingrequests/node-todo-git
  uid: 3413f95b-7cd6-44b7-9a62-daaec834be18
spec:
  backingServiceSelectors:
    - group: etcd.database.coreos.com
      kind: EtcdCluster
      resourceRef: example
      version: v1beta2
  detectBindingResources: true
status:
  bindingStatus: BindingFail
  conditions:
    - lastHeartbeatTime: '2020-03-12T21:05:59Z'
      lastTransitionTime: '2020-03-12T21:05:45Z'
      message: application ResourceRef or MatchLabel not found
      reason: BindingFail
      status: 'False'
      type: Ready
  secret: node-todo-git
```

**Comment by DhritiShikhar** [comment]
@sbose78 Can you explain what is the scenario here?

I can see that there are no `ApplicationSelectors` in your SBR. Why do you expect the status to be `BindingSuccess` in that case?

**Comment by sbose78** [comment]
>  I can see that there are no ApplicationSelectors in your SBR. 

I was trying generic binding. What's the alternative I should be trying?

**Comment by isutton** [comment]
It's more than time to remove the `status.bindingStatus` field.

Before, we couldn't have an empty application and/or empty service fields. This has changed lately, and perhaps it is time to split the `Ready` type into more specific types, for example `CollectionReady` and `InjectionReady`. `CollectionReady` would be related to the collection and persistence of intermediate manifests (secrets, config maps, etc) and `InjectionReady` related to change the applications' manifests to use those intermediate manifests:

```yaml
status:
  conditions:
    - message: application ResourceRef or MatchLabel not found
      reason: IncompleteApplication
      status: 'False'
      type: InjectionReady
    - status: 'True' # secrets, config maps and other intermediate manifests successfully persisted
      type: CollectionReady
```

**Comment by sbose78** [comment]
@DhritiShikhar Could you please share an update on this?

**Comment by DhritiShikhar** [comment]
@sbose78 The fix for #408 needs to go in first so that status is updated atleast.
This is a PR for the same https://github.com/redhat-developer/service-binding-operator/pull/414
